src/sentry/eventtypes/error.py

```diff
@@ -22,42 +21,44 @@ def get_crash_location(data):
 
         func = get_function_name_for_frame(frame, data.get("platform"))
         return frame.get("filename") or frame.get("abs_path"), func
+    return None
+
+
+NATIVE_MOBILE_PLATFORMS = (
+    "cocoa",
+    "objc",
+    "native",
+    "swift",
+    "c",
+    "android",
+    "apple-ios",
+    "cordova",
+    "capacitor",
+    "javascript-cordova",
+    "javascript-capacitor",
+    "react-native",
+    "flutter",
+    "dart-flutter",
+    "unity",
+    "dotnet-xamarin",
+)
 
 
 class ErrorEvent(BaseEvent):
     key = "error"
 
     def extract_metadata(self, data: MutableMapping[str, Any]) -> Metadata:
-
+        """Extracts the metadata from the event data."""
         exceptions = get_path(data, "exception", "values")
         if not exceptions:
             return {}
 
-        # When there are multiple exceptions, we need to pick one to extract the metadata from.
-        # If the event data has been marked with a main_exception_id, then we should be able to
-        # find the exception with the matching metadata.exception_id and use that one.
-        # This can be the case for some exception groups.
-
-        # Otherwise, the default behavior is to use the last one in the list.
-
+        # With chained exceptions, the SDK could set the main_exception_id to
+        # indicate which exception to use for title & subtitle
         main_exception_id = get_path(data, "main_exception_id")
-        exception = (
```

Comment:
Moved this logic into `get_exception`
<img width="632" alt="image" src="https://github.com/getsentry/sentry/assets/44410/95b09d93-863b-4a4d-b60d-83e431c6a3b3">